Library/ViewModels/PledgePaymentMethodsViewModel.swift

```diff
@@ -146,7 +164,12 @@ public final class PledgePaymentMethodsViewModel: PledgePaymentMethodsViewModelT
 
         var selectedPaymentMethod: PaymentSourceSelected?
         if let selectedCardId = selectedCard?.id {
-          selectedPaymentMethod = .savedCreditCard(selectedCardId)
+          if let stripeCardId = selectedCard?.stripeCardId {
+            selectedPaymentMethod = .savedCreditCard(selectedCardId, stripeCardId)
+          } else {
+            assert(false, "Expected stripeCardId to be set. Late pledges may fail if this value is missing.")
```

Comment:
Just confirming: this assert only crashes debug builds? Would we want to log this as an error to firebase for release builds? Up to you if you think that's worth doing or not. (Probably out of scope for this, but I think it'd be nice to have an assert helper function that crashed debug builds and logged to firebase for beta/release builds.)

Reply:
To the best of my knowledge, yes - we have assertions disabled at the compiler level for production builds:
<img width="849" alt="Screenshot 2024-06-10 at 11 54 18 AM" src="https://github.com/kickstarter/ios-oss/assets/146007185/f4d2c1c4-40d9-4916-a373-55793a139cd9">

I loooove the idea of something that has both an assertion and a log to firebase! Let me think about that.